src/Mod/Measure/Gui/ViewProviderMeasureAngle.cpp

```diff
@@ -0,0 +1,340 @@
+ *   version 2 of the License, or (at your option) any later version.      *
+ *                                                                         *
+ *   This library  is distributed in the hope that it will be useful,      *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU Library General Public License for more details.                  *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,         *
+ *   Suite 330, Boston, MA  02111-1307, USA                                *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "PreCompiled.h"
+
+#ifndef _PreComp_
+# include <sstream>
+# include <QApplication>
+# include <Inventor/draggers/SoTranslate2Dragger.h>
+# include <Inventor/nodes/SoAnnotation.h>
+# include <Inventor/nodes/SoBaseColor.h>
+# include <Inventor/nodes/SoCoordinate3.h>
+# include <Inventor/nodes/SoDrawStyle.h>
+# include <Inventor/nodes/SoFontStyle.h>
+# include <Inventor/nodes/SoIndexedLineSet.h>
+# include <Inventor/nodes/SoPickStyle.h>
+# include <Inventor/nodes/SoText2.h>
+# include <Inventor/nodes/SoTranslation.h>
+# include <Inventor/engines/SoCalculator.h>
+# include <Inventor/engines/SoComposeVec3f.h>
+# include <Inventor/engines/SoConcatenate.h>
+# include <Inventor/engines/SoComposeMatrix.h>
+# include <Inventor/engines/SoComposeRotation.h>
+# include <Inventor/engines/SoComposeRotationFromTo.h>
+# include <Inventor/engines/SoDecomposeRotation.h>
+# include <Inventor/engines/SoTransformVec3f.h>
+# include <Inventor/nodekits/SoShapeKit.h>
+# include <Inventor/nodes/SoFont.h>
+# include <Inventor/nodes/SoLineSet.h>
+# include <Inventor/nodes/SoMatrixTransform.h>
+# include <Inventor/nodes/SoSeparator.h>
+# include <Inventor/nodes/SoTransform.h>
+# include <Inventor/nodes/SoVertexProperty.h>
+# include <Inventor/nodekits/SoBaseKit.h>
+#endif
+
+#include <Precision.hxx>
+#include <Geom_Curve.hxx>
+#include <Geom_Line.hxx>
+#include <gp_Vec.hxx>
+#include <gp_Lin.hxx>
+#include <gp_Pnt.hxx>
+# include <GeomAPI_ExtremaCurveCurve.hxx>
+# include <GeomAPI_ProjectPointOnCurve.hxx>
+
+#include <App/Application.h>
+#include <App/Document.h>
+#include <Base/Console.h>
+#include <Base/Exception.h>
+#include <Base/Quantity.h>
+#include <Gui/ArcEngine.h>
+#include <Gui/Command.h>
+#include <Gui/Document.h>
+#include <Gui/ViewParams.h>
+
+#include <Mod/Measure/App/Preferences.h>
+
+#include "ViewProviderMeasureAngle.h"
+
+
+using namespace MeasureGui;
+using namespace Measure;
+
+gp_Lin* getLine(gp_Vec& vec, gp_Vec& origin) {

```

Comment:
Why does this return a pointer? In the code below it's used like this:
`gp_Lin lin1 = *getLine(vector1, loc1);` that causes a memory leak.